crates/apollo-parser/src/ast/node_ext.rs

```diff
@@ -281,3 +436,133 @@ fn text_of_first_token(node: &SyntaxNode) -> TokenText {
 
     TokenText(first_token)
 }
+
+#[cfg(test)]
+mod string_tests {
+    use super::unescape_string;
+
+    #[test]
+    fn it_parses_strings() {
+        assert_eq!(unescape_string(r"simple"), "simple");
+        assert_eq!(unescape_string(r" white space "), " white space ");
+    }
+
+    #[test]
+    fn it_unescapes_strings() {
+        assert_eq!(unescape_string(r#"quote \""#), "quote \"");
+        assert_eq!(
+            unescape_string(r"escaped \n\r\b\t\f"),
+            "escaped \n\r\u{0008}\t\u{000c}"
+        );
+        assert_eq!(unescape_string(r"slashes \\ \/"), r"slashes \ /");
+        assert_eq!(
+            unescape_string("unescaped unicode outside BMP \u{1f600}"),
+            "unescaped unicode outside BMP \u{1f600}"
+        );
```

Comment:
This test case is misleading. It looks like it’s testing for https://github.com/apollographql/apollo-rs/issues/640 but that’s not implemented yet. The test passes because the Rust input string is not a raw string so `\u{1f600}` is unescaped by the Rust parser and `unescape_string` doesn’t end up unescaping anything in this case.